src/components/Home.jsx

```diff
@@ -1,11 +1,35 @@
+import { useEffect, useState } from "react";
 import AllArticles from "./AllArticles";
 import SortAndFilter from "./SortAndFilter";
+import ArticlesByTopic from "./ArticlesByTopic";
+import { getTopics } from "../utils/apiRequests";
+import { useSearchParams } from "react-router-dom";
 
 const Home = () => {
-  return (
-    <main id="home">
-      <SortAndFilter /> <AllArticles />
-    </main>
-  );
+  const [orderQueries, setOrderQueries] = useState([]);
+  const [params] = useSearchParams();
+  const topic = params.get("topic");
+  const [topics, setTopics] = useState([]);
+
+  useEffect(() => {
+    getTopics().then(({ data }) => {
+      setTopics(data.topics);
+    });
+  }, []);
+  if (topic === null) {
+    return (
+      <main id="home">
+        <SortAndFilter topics={topics} setOrderQueries={setOrderQueries} />{" "}
+        <AllArticles orderQueries={orderQueries} />
+      </main>
+    );
+  } else {
+    return (
+      <main id="home">
+        <SortAndFilter topics={topics} setOrderQueries={setOrderQueries} />{" "}
+        <ArticlesByTopic topic={topic} orderQueries={orderQueries} />
```

Comment:
If the only difference here is whether you render `AllArticles` or `ArticlesByTopic` then consider doing that inline rather than repeating the rest of the elements. 

e.g. `{topic === null ? <AllArticles /> : <ArticlesByTopic />}`